src/index.js

```diff
@@ -174,29 +320,28 @@ module.exports = class WyzeAPI {
   }
 
   async login() {
-    let result;
-    // Do we need apiKey or keyId?
+    // Wyze requires 2 sets of credentials to login + MFA - username/password, apiKey/keyId
     if (this.apiKey == null) {
       throw new Error(
         'ApiKey Required, Please provide the "apiKey" parameter in config.json'
       );
     } else if (this.keyId == null) {
       throw new Error(
-        'KeyId Required, Please provide the "keyid" parameter in config.json'
+        'KeyId Required, Please provide the "keyId" parameter in config.json'
       );
     } else {
-      try {
-        result = await this._performLoginRequest();
-        if (this.apiLogEnabled)
-          this.log.debug("Successfully logged into Wyze API");
-        await this._updateTokens(result.data);
-      } catch (error) {
+      const result = await this._performLoginRequest();
+      if (!result.ok || !result.data.access_token) {
         throw new Error(
-          "Invalid credentials, please check username, password, keyid or apikey" + error
-        );
+            result
+          )}
       }
+
+      if (this.apiLogEnabled) {
+        this.log.debug("Successfully logged into Wyze API");
+      }
+      await this._updateTokens(result.data);
```

Comment:
ReferenceError: result is not defined.

Reply:
my bad - I had a pending change on my local

![image](https://github.com/jfarmer08/wyze-api/assets/23192126/5e0dca88-61aa-489a-95bc-16457392730a)